File: Java/1137. N-th Tribonacci Number.java
class Solution {
    public int tribonacci(int n) {
        if(n < 3){
            return n == 0 ? 0 : 1;
        }
        
        int n1 = 0, n2 = 1, n3 = 1;
        for(int i=0; i<n-2; i++){
            int temp = n1 + n2 + n3;
            n1 = n2;
            n2 = n3;
            n3 = temp;
        }
        
        return n3;
    }
}

File: Python3/1137. N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        n1, n2, n3 = 0, 1, 1
        if n == 0:
            return n1
        elif n == 1 or n == 2:
            return n2
        
        for i in range(n-2):
            n1, n2, n3 = n2, n3, n1+n2+n3
            
        return n3